### PR TITLE
cpu/mpk1839/kl1839vm1.cpp: Improved VAX-11 code decoder in user space

### PR DESCRIPTION
**Pull request by holub**
Good progress here. CPU is able to decode/process VAX-11 instructions in user space.
image
image

### CODE REVIEW

File: src/devices/cpu/mpk1839/kl1839vm1.cpp [hunk 1]
```diff
@@ -5,22 +5,26 @@
...
 #include "kl1839vm1.h"
 #include "kl1839vm1dasm.h"
+#include "cpu/vax/vaxdasm.h"
+#include <regex>
 
 
-#define VERBOSE ( LOG_GENERAL )
+#define LOG_VAX    (1U << 1)
+
+#define VERBOSE ( LOG_GENERAL | LOG_VAX )
```
Comment (rb6502) [inline src/devices/cpu/mpk1839/kl1839vm1.cpp:14]: Don't check in with VERBOSE enabled.

Comment (holub) [reply src/devices/cpu/mpk1839/kl1839vm1.cpp:14]: Is it ok to keep LOG_GENERAL?

Comment (holub) [reply src/devices/cpu/mpk1839/kl1839vm1.cpp]: done

File: src/devices/cpu/mpk1839/kl1839vm1.h [hunk 2]
```diff
@@ -75,17 +95,24 @@ class kl1839vm1_device :  public cpu_device
...
+	PAIR            m_amc;    // Microdode PC
+	PAIR            m_ppc;    // previous program counter
+	bool            m_fp;
+	bool            m_jzdra_waiting;
+	u32             m_consts[0x10] = { 0x4, 0x2, 0x8, 0x1, 0x0, 0, 0, 0x66, 0, 0xc00000, 0xffffffff, 0x1f0000, 0x4000000, 0, 0, 0 };
+	PAIR            m_reg[0x20];
+	int             m_icount;
+	u32             m_op_size;
+
+	std::deque<u32> m_pcm_queue;
```
Comment (cuavas) [inline src/devices/cpu/mpk1839/kl1839vm1.h:114]: You’ll need to change this to something else to support save states.  You can’t register a `std::deque` for save states.

**Pull request merged**